src/test/shell/bazel/run_test.sh

```diff
@@ -125,4 +125,90 @@ eof
   echo "$output" | grep --fixed-strings 'ExecuteProgram(C:\first_part second_part)' || fail "Expected error message to contain unquoted path"
 }
 
+function test_run_with_runfiles_env() {
```

Comment:
This test case is not in the original PR: https://github.com/bazelbuild/bazel/pull/18498/files

Did you have to deal with some merge conflict? Let's remove this one since it's unrelated.